### PR TITLE
update iceberg procedures according to #22608

### PR DESCRIPTION
**Pull request by jackychen718**
According to #22608:
Add subheading Procedures, put Register table, Unregister table, Rollback to snapshots, Expire snapshots under the subheading.

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
      INSERT    |       2 |  677209275408372885 | {orderkey=18016, custkey=403, orderstatus=O, totalprice=174070.99, orderdate=1996-03-19, orderpriority=1-URGENT, clerk=Clerk#000000629, shippriority=0, comment=ly. quickly ironic excuses are furiously. carefully ironic pack}
      INSERT    |       2 |  677209275408372885 | {orderkey=18017, custkey=958, orderstatus=F, totalprice=203091.02, orderdate=1993-03-26, orderpriority=1-URGENT, clerk=Clerk#000000830, shippriority=0, comment=sleep quickly bold requests. slyly pending pinto beans haggle in pla}
 
+Procedures
+----------
+
+Use the CALL statement to perform data manipulation or administrative tasks.Procedures are available in the system schema of each catalog iceberg. The following are several examples:
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:631]:
```suggestion
Use the CALL statement to perform data manipulation or administrative tasks. Procedures are available in the ``system`` schema of the catalog. 
```

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+
+Register table
+^^^^^^^^^^^^^^
+
+Iceberg tables for which table data and metadata already exist in the
+file system can be registered with the catalog using the ``register_table``
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:637]:
```suggestion
file system can be registered with the catalog. Use the ``register_table``
```
Suggest breaking up this single sentence paragraph to improve readability.

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+procedure on the catalog's ``system`` schema by supplying the target schema,
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:638]:
```suggestion
procedure on the catalog's ``system`` schema and supply the target schema,
```
Minor change to support new sentence creation in the previous line.

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+
+A metadata file can optionally be included as an argument to ``register_table``
+in the case where a specific metadata file contains the targeted table state::
+
+    CALL iceberg.system.register_table('schema_name', 'table_name', 'hdfs://localhost:9000/path/to/iceberg/table/metadata/dir', '00000-35a08aed-f4b0-4010-95d2-9d73ef4be01c.metadata.json')
+
+.. note::
+
+    When registering a table with the Hive metastore, the user calling the
+    procedure will be set as the owner of the table and will have ``SELECT``,
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:656]:
```suggestion
    procedure is set as the owner of the table and will have ``SELECT``,
```

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for doing this! A few small nits that could be addressed while these topics are being reorganized.

File: presto-docs/src/main/sphinx/connector/iceberg.rst (hunk 1 above)
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:631]: OK

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 5]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+
+    Table data and metadata will remain in the filesystem after a call to
+    ``unregister_table`` only when using the Hive catalog. This is similar to
+    the behavior listed above for the ``DROP TABLE`` command.
+
+
+Rollback to snapshots
+^^^^^^^^^^^^^^^^^^^^^
+
+Roll back a table to a specific snapshot ID. Iceberg can roll back to a specific snapshot ID by using ``rollback_to_snapshot`` procedure on iceberg`s ``system`` schema::
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:683]: OK

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 6]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+    CALL iceberg.system.unregister_table('schema_name', 'table_name')
+
+.. note::
+
+    Table data and metadata will remain in the filesystem after a call to
+    ``unregister_table`` only when using the Hive catalog. This is similar to
+    the behavior listed above for the ``DROP TABLE`` command.
+
+
+Rollback to snapshots
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:680]: OK

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 7]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+    procedure will be set as the owner of the table and will have ``SELECT``,
+    ``INSERT``, ``UPDATE``, and ``DELETE`` privileges for that table. These
+    privileges can be altered using the ``GRANT`` and ``REVOKE`` commands.
+
+.. note::
+
+    When using the Hive catalog, attempts to read registered Iceberg tables
+    using the Hive connector will fail.
+
+Unregister table
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:665]: OK

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 8]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
      INSERT    |       2 |  677209275408372885 | {orderkey=18016, custkey=403, orderstatus=O, totalprice=174070.99, orderdate=1996-03-19, orderpriority=1-URGENT, clerk=Clerk#000000629, shippriority=0, comment=ly. quickly ironic excuses are furiously. carefully ironic pack}
      INSERT    |       2 |  677209275408372885 | {orderkey=18017, custkey=958, orderstatus=F, totalprice=203091.02, orderdate=1993-03-26, orderpriority=1-URGENT, clerk=Clerk#000000830, shippriority=0, comment=sleep quickly bold requests. slyly pending pinto beans haggle in pla}
 
+Procedures
+----------
+
+Use the CALL statement to perform data manipulation or administrative tasks.Procedures are available in the system schema of each catalog iceberg. The following are several examples:
+
+Register table
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:633]: OK

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 9]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+
+===================== ========== =============== =======================================================================
+Argument Name         required   type            Description
+===================== ========== =============== =======================================================================
+``table``             ✔️          string          Name of the table to update
+
+``snapshot_id``       ✔️          long            Snapshot ID to rollback to
+===================== ========== =============== =======================================================================
+
+Expire snapshots
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:697]: ok

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

Pull updated branch, new local docs build, verified that all requested changes by myself and @ZacBlanco have been made in the current file.

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 10]
```diff
@@ -625,6 +625,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+    CALL iceberg.system.unregister_table('schema_name', 'table_name')
+
+.. note::
+
+    Table data and metadata will remain in the filesystem after a call to
+    ``unregister_table`` only when using the Hive catalog. This is similar to
+    the behavior listed above for the ``DROP TABLE`` command.
+
+
+Rollback to Snapshots
```
Comment (ZacBlanco) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:680]:
```suggestion
Rollback to Snapshot
```

Sorry to be pedantic - I think it should match the procedure name

**Review by ZacBlanco — CHANGES_REQUESTED** [review]

**Comment by hantangwangd** [comment]
@jackychen718 Thanks for the fix. Please use rebase rather than merge, in that way we can remove the merge commit.

File: presto-docs/src/main/sphinx/connector/iceberg.rst (hunk 10 above)
Comment (steveburnett) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:682]: When this is edited, also edit the next line "^^^^^^^^^^^^^^^^^^^^^" to match the length of the characters in the text of the heading

**Comment by jackychen718** [comment]
@hantangwangd Sorry for asking. I see rebase is used to merge an branch without unneccessary commit. Locally, I only have one master branch. After modify the file, could you please give some more guidance how to use rebase in the scenario.Thanks a lot.

**Comment by hantangwangd** [comment]
@jackychen718 In order to use rebase, you need a dedicated branch to synchronize with `prestodb:master`, then rebase your working branch which contains new commits onto it. Let's call these two branches `newest` and `master` for example. You can follow the following steps to realize a rebase:

1. git checkout -B newest	// create a new branch `newest` based on your current working branch

2. git reset --hard HEAD~2	// hard reset all your own commits, assuming there are 2

3. git push -f origin newest	// push `newest` to remote repository, use `git remote -v` to get your remote repository name, assuming the name is `origin`

4. Then select branch `newest` in your own fork repository on github, and click `Sync fork` -> `update branch` to synchronize with `prestodb:master`

5. git fetch origin -> git rebase origin/newest

In this way, you will rebase you working branch `master` onto the newest commit list without a merge. Hoping this can help you, any questions please let me know.

**Comment by jackychen718** [comment]
OK. I created another PR as described above with rebase command. Should this PR be closed?

**Comment by hantangwangd** [comment]
> OK. I created another PR as described above with rebase command. Should this PR be closed?

Sure, you can close it.

**Pull request closed without merging**